checker/checker_test.go

```diff
@@ -297,15 +305,229 @@ func TestChecker_Check(t *testing.T) {
 				Name: "myFunction",
 			},
 		},
-	}} {
+	}, {
+		"variadic options with bad type",
+		`
+		fs default() {
+			myfunc "string"
+		}
+		fs myfunc(variadic option::run opts) {
+			image "busybox"
+			run "echo hi" with opts
+		}
+		`,
+		ErrWrongArgType{
+			Pos:      makePos(2, 8),
```

Comment:
Do you think we should abstract the testing away from these node positions? Or do you think they belong here?